tests/integration/ipdevpoll/plugins/snmpcheck_test.py

```diff
@@ -21,8 +21,17 @@ def test_short_outage(localhost, db):
     job._create_agentproxy = Mock()
     job._destroy_agentproxy = Mock()
     agent.walk.return_value = defer.succeed(False)
+    event_count_start = EventQueue.objects.filter(
+        source_id='ipdevpoll',
+        target_id='eventEngine',
+        event_type='snmpAgentState',
+        netbox_id=localhost.pk,
+        state=EventQueue.STATE_START,
+    ).count()
```

Comment:
It does in fact look like this test now calls the same filter *three* times. This could really benefit from a cleanup to remove redundancies.

Remember that Django query sets will only actually fetch results from the database when needed, so this whole `EventQueue` query could be defined once as a "template", and its `count()` method only called when a new count of the matches is needed.

e.g.

```suggestion
    start_event = EventQueue.objects.filter(
        source_id='ipdevpoll',
        target_id='eventEngine',
        event_type='snmpAgentState',
        netbox_id=localhost.pk,
        state=EventQueue.STATE_START,
    )
    event_count_start = start_event.count()
```

Then later

```python
    assert start_event.count() > event_count_start
```